### PR TITLE
Issue 61 3 d trafo show differences start target

### PR DESCRIPTION
**Pull request by TWoerner94**

### CODE REVIEW

File: client/src/actions/submitCoords/submitCoordsActions.js [hunk 1]
```diff
@@ -64,4 +68,55 @@ export const submitCoords = () => (dispatch, getState) => {
...
+ */
+export const calculateDifference = (startPoints, targetPoints, trafoParams) => (dispatch, getState) => {
+  dispatch(calculateDifferenceRequest());
+  return axios.post('/calculate-trafo-difference', {
+      startPoints,
+      targetPoints,
+      trafoParams,
+    })
+    .then(response => {
+      if (response.data[0].hasOwnProperty('vx')) {
```
Comment (brauls) [inline client/src/actions/submitCoords/submitCoordsActions.js]: mabe check here first, wether the data-list is empty ?

File: client/src/components/BarGraph/BarGraph.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                beginAtZero:true
+              }
+            }]
+          }
+        }}
+      />
+    </div>
+  )
+}
+
```
Comment (brauls) [inline client/src/components/BarGraph/BarGraph.js]: maybe look at Props type checking (PropTypes): https://reactjs.org/docs/typechecking-with-proptypes.html
-> must not be implemented in this PR, but may be a new Issue

File: client/src/actions/submitCoords/submitCoordsActions.js [hunk 3]
```diff
@@ -3,6 +3,10 @@ export const SUBMIT_COORDS = 'SUBMIT_COORDS';
 export const SUBMIT_COORDS_REQUEST = 'SUBMIT_COORDS_REQUEST';
 export const SUBMIT_COORDS_SUCCESS = 'SUBMIT_COORDS_SUCCESS';
 export const SUBMIT_COORDS_FAILURE = 'SUBMIT_COORDS_FAILURE';
+export const CALCULATE_DIFFERENCE = 'CALCULATE_DIFFERENCE';
```
Comment (martiL) [inline client/src/actions/submitCoords/submitCoordsActions.js]: What kind of differences ?

File: client/src/containers/ThreeDTrafoResultContainer.js [hunk 4]
```diff
@@ -3,16 +3,31 @@ import { connect }          from 'react-redux';
...
+}                           from '../selectors/TrafoSelectors/getTrafoInputDataSelector/getTrafoInputDataSelector';
+import {
+  calculateDifference
+}                           from '../actions/submitCoords/submitCoordsActions';
 import { removeError }      from '../actions/errorHandling/errorHandlingActions';
 import ThreeDTrafoResult    from '../components/ThreeDTrafoResult/ThreeDTrafoResult';
 
 const mapDispatchToProps = dispatch => ({
   onRemoveError: () => dispatch(removeError()),
+  onCalculateDifference: (startPoints, targetPoints, trafoParams) => dispatch(calculateDifference(startPoints, targetPoints, trafoParams)),
```
Comment (martiL) [inline client/src/containers/ThreeDTrafoResultContainer.js]: name is to general

File: client/src/containers/ThreeDTrafoResultContainer.js [hunk 5]
```diff
@@ -48,6 +80,7 @@ class ThreeDTrafoResultContainer extends Component {
       <div>
         <ThreeDTrafoResult
           trafoParams={ this.props.trafoParams }
+          trafoDifference={ this.props.trafoDifference }
           error={ this.props.error }
           isCalculating = { this.props.isCalculating }
```
Comment (martiL) [inline client/src/containers/ThreeDTrafoResultContainer.js]: isCalculating and error should be handled in the conatiner not the component

File: server.js [hunk 6]
```diff
@@ -90,6 +90,59 @@ app.post('/param-inversion', function(req, res) {
...
 
+
+app.post('/calculate-trafo-difference', function(req, res) {
+  const startPoints = req.body.startPoints;
+  const targetPoints = req.body.targetPoints;
+  const trafoParams = req.body.trafoParams;
+
+  let calculatedTarget = [];
+  let differences = [];
+  let objDifference = {
```
Comment (martiL) [inline server.js]: maybe we should start using the npm module -> https://github.com/sigma3Dev/s3d-fitting-commands

File: client/src/actions/submitCoords/submitCoordsActions.js (hunk 1 above)
Comment (TWoerner94) [reply client/src/actions/submitCoords/submitCoordsActions.js]: done

**Pull request merged**